### PR TITLE
Replace all JITaaS prefix with JITServer under runtime folder

### PR DESCRIPTION
**Pull request by a7ehuo**
As decided by eclipse/openj9#6203, replace all `JITaaS` prefix with `JITServer` under `runtime`  folder.

Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

File: runtime/compiler/control/MethodToBeCompiled.cpp [hunk 1]
```diff
@@ -96,7 +96,7 @@ void TR_MethodToBeCompiled::initialize(TR::IlGeneratorMethodDetails & details, v
 void
 TR_MethodToBeCompiled::shutdown()
    {
-   cleanupJITaaS();
+   cleanupJITServer();
```
Comment (mpirvu) [inline runtime/compiler/control/MethodToBeCompiled.cpp]: In another PR the name of this method has been changed

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
Sanity testing only shows the known `cmdLineTester_gcsuballoctests_0` test failure

File: runtime/compiler/control/MethodToBeCompiled.cpp (hunk 1 above)
Comment (a7ehuo) [reply runtime/compiler/control/MethodToBeCompiled.cpp]: #6893 changes `cleanupJITaaS` to `freeJITServerAllocations`. I assume this PR will be merged first. Then I'll do rebase for PR #6893. This PR is dedicated to replace all `jitaas` names.

File: runtime/compiler/control/CompilationThread.cpp [hunk 2]
```diff
@@ -5336,7 +5336,7 @@ void TR::CompilationInfo::recycleCompilationEntry(TR_MethodToBeCompiled *entry)
    entry->_freeTag |= ENTRY_IN_POOL_NOT_FREE;
    if (entry->_numThreadsWaiting == 0)
       entry->_freeTag |= ENTRY_IN_POOL_FREE;
-   entry->cleanupJITaaS();
+   entry->cleanupJITServer();
```
[the remark this answers is not in the document]
Comment (a7ehuo) [reply runtime/compiler/control/CompilationThread.cpp]: #6893 changes cleanupJITaaS to freeJITServerAllocations. I assume this PR will be merged first. Then I'll do rebase for PR #6893. This PR is dedicated to replace all jitaas names.

**Comment by a7ehuo** [comment]
@mpirvu All comments addressed. Ready for another review. Thanks.

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8,jdk11 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
For Java8 the only error is the known `cmdLineTester_gcsuballoctests_0` test
For Java11 there is also this: `J9vmTest_2`
```
+++ j9vm.test.jni.CriticalRegionTest: +++
command: /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/java -XX:+UseJITServer -Xcompressedrefs -Xgcpolicy:metronome -Xcompressedrefs  -Xdump -Xms64m -Xmx64m    -classpath /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../../jvmtest/functional/VM_Test/VM_Test.jar:/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../../jvmtest/TestConfig/lib/asm-all.jar  j9vm.test.jni.CriticalRegionTest 

Exception in thread "main" java.lang.BootstrapMethodError: java.lang.invoke.StringConcatException: Generator failed
	at j9vm.test.jni.CriticalRegionTest.testAcquireAndSleep(CriticalRegionTest.java:756)
	at j9vm.test.jni.CriticalRegionTest.runBlockingTests(CriticalRegionTest.java:733)
	at j9vm.test.jni.CriticalRegionTest.main(CriticalRegionTest.java:191)
Caused by: java.lang.invoke.StringConcatException: Generator failed
	at java.base/java.lang.invoke.StringConcatFactory.generate(StringConcatFactory.java:764)
	at java.base/java.lang.invoke.StringConcatFactory.doStringConcat(StringConcatFactory.java:665)
	at java.base/java.lang.invoke.StringConcatFactory.makeConcatWithConstants(StringConcatFactory.java:581)
	at java.base/java.lang.invoke.MethodHandle.invokeBsm(MethodHandle.java:963)
	at java.base/java.lang.invoke.MethodHandle.resolveInvokeDynamic(MethodHandle.java:1093)
	... 3 more
Caused by: java.lang.ClassCastException: [I incompatible with java.util.concurrent.ConcurrentHashMap$Node
	at java.base/java.util.concurrent.ConcurrentHashMap.tabAt(ConcurrentHashMap.java:760)
	at java.base/java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:938)
	at java.base/java.lang.invoke.ThunkTable.get(MethodHandle.java:1704)
	at java.base/java.lang.invoke.FoldHandle.computeThunks(FoldHandle.java:69)
	at java.base/java.lang.invoke.MethodHandle.<init>(MethodHandle.java:235)
	at java.base/java.lang.invoke.FoldHandle.<init>(FoldHandle.java:33)
	at java.base/java.lang.invoke.FoldNonvoidHandle.<init>(FoldHandle.java:104)
	at java.base/java.lang.invoke.FoldHandle.get(FoldHandle.java:55)
	at java.base/java.lang.invoke.MethodHandles.foldArgumentsCommon(MethodHandles.java:2721)
	at java.base/java.lang.invoke.MethodHandles.foldArguments(MethodHandles.java:2637)
	at java.base/java.lang.invoke.StringConcatFactory$MethodHandleInlineCopyStrategy.generate(StringConcatFactory.java:1607)
	at java.base/java.lang.invoke.StringConcatFactory.generate(StringConcatFactory.java:756)
	... 7 more
no-zero exit value: 1
*** Test FAILED *** (j9vm.test.jni.CriticalRegionTest)
```

It's quite possible that this error already existed.

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk11 depends eclipse/openj9-omr#jitaas

**Comment by a7ehuo** [comment]
Java 8 and Java 11 test failed with the known test `cmdLineTester_gcsuballoctests_0`. 

Will do another rebase to resolve the latest conflict. 

[Java 11](https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/196/console)
```
14:48:19  FAILED test targets:
14:48:19  	cmdLineTester_gcsuballoctests_0
14:48:19  
14:48:19  TOTAL: 226   EXECUTED: 138   PASSED: 137   FAILED: 1   DISABLED: 4   SKIPPED: 84
```
[Java 8](https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal/224/console)
```
05:18:14  FAILED test targets:
05:18:14  	cmdLineTester_gcsuballoctests_0
05:18:14  
05:18:14  TOTAL: 204   EXECUTED: 117   PASSED: 116   FAILED: 1   DISABLED: 6   SKIPPED: 81
```

**Comment by a7ehuo** [comment]
@mpirvu The latest conflict is resolved.

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8,jdk11 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8,jdk11 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
jdk11 had a failure in `cmdLineTester_decompilationTests_1` which must be intermittent because we didn't see this in previous testing with this PR:
```
Testing: decomp003-FSD-DECOMP-DURING-EVENT
Test start time: 2019/09/04 03:56:34 Atlantic Standard Time
Running command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/java" -XX:+UseJITServer -Xcompressedrefs -Xcompressedrefs -Xgcpolicy:gencon -Xjit:enableOSR,count=0  -Xdump -Xgcthreads1  -Xdump:system:events=abort  -Xjit:count=0,limit={*.jit*},tryToInline={*.jit*} -agentlib:jvmtitest=test:decomp003,args:stepincatch -cp "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitest.jar" com.ibm.jvmti.tests.util.TestRunner
Time spent starting: 1 milliseconds
Time spent executing: 33231 milliseconds
Test result: FAILED
Output from test:
 [OUT] *** Testing [1/1]:	testExceptions
 [OUT] Successfully caught exception in outer frame
 [OUT] JVMTI detected correct catcher
 [OUT] Successfully caught exception in shallowest inline frame
 [OUT] JVMTI detected correct catcher
 [ERR] 06:57:02.952 0x12ee200    j9vm.249    *   ** ASSERTION FAILED ** at BytecodeInterpreter.hpp:8016: ((0 ))
```
Java8 only shows the known `cmdLineTester_gcsuballoctests_0 ` failure.
All in all this PR only performs some name changes, so it shouldn't cause test failures.

**Pull request merged**